src/server/trace/viewer/traceViewer.ts

```diff
@@ -196,6 +196,21 @@ async function appendTraceEvents(model: TraceModel, file: string) {
 }
 
 export async function showTraceViewer(tracePath: string, browserName: string, headless = false): Promise<BrowserContext | undefined> {
+  const dir = fs.mkdtempSync(path.join(os.tmpdir(), `playwright-trace`));
+  process.on('exit', () => rimraf.sync(dir));
+
+  if (/^https?:\/\//i.test(tracePath)){
+    const downloadZipPath = path.join(dir, 'trace.zip');
+    const {error} = await downloadFile(tracePath, downloadZipPath, {
```

Comment:
Why don't we repurpose `downloadFile` into `downloadFileWith(Optional?)ProgressBar` instead of exposing inner utilities like `getDownloadProgress`? There are not many usages anyway.

Reply:
Will do that.
Should I extract out the retry logic so that zip files will also retry 3 times in case of errors? 

https://github.com/microsoft/playwright/blob/0908dc98c8d6226fa02c6a0a5fd81003dcb8e158/src/utils/browserFetcher.ts#L56-L73

Reply:
Sounds good!